scripts/analysis.py

```diff
@@ -0,0 +1,378 @@
+
+    nuclide_set = set(nuclides)
+    mass_dict = collections.OrderedDict({})
+
+    for nuclide in nuclide_set:
+        temp_nuclide_sum = 0 
+        for i in range(len(nuclides)):
+            if nuclides[i] == nuclide:
+                temp_nuclide_sum += mass_of_nuclides[i]
+        mass_dict[nuclide] = temp_nuclide_sum
+
+
+    for nuclide in mass_dict:
+        nuclide_name = str(nucname.name(nuclide))
+        nuclide_inven += nuclide_name + ' = ' + str(mass_dict[nuclide]) + 'kg \n'
+
+    return nuclide_inven
+
+
+def capacity_calc(governments, timestep, entry, exit):
+    """ Adds and subtracts capacity over time for plotting
+
+    Paramters
+    ---------
+    governments: array
+        array of governments (countries)
+    timestep: array
+        array of timestep from 0 to simulation time
+    entry: array
+        power_cap, agentid, parentid, entertime
+        of all entered reactors
+    exit: array
+        power_cap, agentid, parenitd, exittime
+        of all decommissioned reactors
+
+    Returns
+    -------
+    power_dict: dictionary
+        dictionary of capacity progression with country_government as key
+    num_dict: dictionary
+        dictionary of number of reactors progression
+        with country_government as key
+    """
+
+    power_dict = collections.OrderedDict({})
+    num_dict = collections.OrderedDict({})
+
+    for gov in governments:
+        capacity = []
+        num_reactors = []
+        cap = 0
+        count = 0
+        for num in timestep:
+            for enter in entry:
+                if enter[3] == num and enter[2] == gov[1]:
+                    cap += enter[0]
+                    count += 1
+            for dec in exit:
+                if dec[3] == num and dec[2] == gov[1]:
+                    cap -= dec[0]
+                    count -= 1
+            capacity.append(cap)
+            num_reactors.append(count)
+
+        power_dict[gov[0]] = np.asarray(capacity)
+        num_dict[gov[0]] = np.asarray(num_reactors)
+
+    return power_dict, num_dict
+
+
+def stacked_bar_chart(dictionary, timestep, xlabel, ylabel, title, outputname):
+    """ Creates stacked bar chart of timstep vs dictionary
+
+    Parameters
+    ----------
+    dictionary: dictionary
+        holds time series data
+    timestep: array
+        array of timestep (x axis)
+    xlabel: string
+        xlabel of plot
+    ylabel: string
+        ylabel of plot
+    title: string
+        title of plot
+
+    Returns
+    -------
+    Stacked bar chart
+    """
+
+    # set different colors for each bar
+    color_index = 0
+    top_index = True
+    prev = ''
+    plot_array = []
+    # for every country, create bar chart with different color
+    for key in dictionary:
+        label = key.replace('_government', '')
+        # very first country does not have a 'bottom' argument
+        if top_index is True:
+            plot = plt.bar(1950+(timestep/12), dictionary[key], .5,
```

Comment:
For readabilitiy, consider breaking at each comma in a long function call. Related, consider including the keyword, even if not necessary, so that it's clear to your future self what .5 is for. 

```python
plot = plt.bar(left=1950+(timestep/12), 
               height=dictionary[key], 
               width=0.5, 
               color=cm.viridis(1.*color_index/len(dictionary)),
               edgecolor='none', 
               label=label)
```